pkgs/development/python-modules/keyring/default.nix

```diff
@@ -1,43 +1,47 @@
 { lib, stdenv, buildPythonPackage, fetchPypi, isPy27, pythonOlder
-, dbus-python
-, entrypoints
-, importlib-metadata
-, pytest
-, pytest-flake8
-, secretstorage
 , setuptools_scm
 , toml
+, importlib-metadata
+, dbus-python
+, jeepney
+, secretstorage
+, pytestCheckHook
 }:
 
 buildPythonPackage rec {
   pname = "keyring";
-  version = "21.5.0";
+  version = "23.0.0";
   disabled = isPy27;
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "207bd66f2a9881c835dad653da04e196c678bf104f8252141d2d3c4f31051579";
+    sha256 = "237ff44888ba9b3918a7dcb55c8f1db909c95b6f071bfb46c6918f33f453a68a";
   };
 
   nativeBuildInputs = [
     setuptools_scm
     toml
   ];
 
-  checkInputs = [ pytest pytest-flake8 ];
+  checkInputs = [
+    pytestCheckHook
+  ];
 
-  propagatedBuildInputs = [ dbus-python entrypoints ]
-  ++ lib.optional stdenv.isLinux secretstorage
-  ++ lib.optionals (pythonOlder "3.8") [ importlib-metadata ];
+  propagatedBuildInputs = [
+    importlib-metadata
```

Comment:
The pythonOlder got lost.

Reply:
https://github.com/jaraco/keyring/blob/v23.0.0/setup.cfg#L28

Reply:
The Package is integrated in 3.8 so upstream should correct their requires.